### PR TITLE
refactor: authz_api_sources swagger spec

### PR DESCRIPTION
**Pull request by JimMoen**

### CODE REVIEW

File: apps/emqx_authz/src/emqx_authz_api_schema.erl [hunk 1]
```diff
@@ -16,490 +16,185 @@
...
+           , {method, #{ type => enum([get, post])
+                       , default => get
+                       , converter => fun to_bin/1}}
+           , {headers, fun headers/1}
+           , {body, fun body/1}
+           , {request_timeout, mk_duration("Request timeout", #{default => "30s"})}]
+        ++ maps:to_list(maps:without([ base_url
+                                     , pool_type],
+                                     maps:from_list(emqx_connector_http:fields(config))));
+fields('built-in-database') ->
```
Comment (DDDHuang) [inline apps/emqx_authz/src/emqx_authz_api_schema.erl:39]: maybe a better name? 
```
'built_in_database'
```

File: apps/emqx_authz/src/emqx_authz_api_schema.erl [hunk 2]
```diff
@@ -16,490 +16,185 @@
...
+    maps:put(<<"content-type">>,
+             <<"application/json">>,
+             default_headers_no_content_type()).
+
+default_headers_no_content_type() ->
+    #{ <<"accept">> => <<"application/json">>
+     , <<"cache-control">> => <<"no-cache">>
+     , <<"connection">> => <<"keep-alive">>
+     , <<"keep-alive">> => <<"timeout=5">>
+     }.
```
Comment (DDDHuang) [inline apps/emqx_authz/src/emqx_authz_api_schema.erl:116]:
```http
Keep-Alive: timeout=30, max=1000
```

File: apps/emqx_authz/src/emqx_authz_api_schema.erl [hunk 3]
```diff
@@ -16,490 +16,185 @@
...
+    ++ [ {rules, #{ type => string()
+                  , example =>
+                        <<"{allow,{username,\"^dashboard?\"},","subscribe,[\"$SYS/#\"]}.\n",
+                          "{allow,{ipaddr,\"127.0.0.1\"},all,[\"$SYS/#\",\"#\"]}.">>}}
+         %% The path will be deprecated, `acl.conf` will be fixed in subdir of `data`
+       , {path, #{ type => string()
+                 , example => <<"acl.conf">>}}];
+fields(position) ->
+    [ { position
+      , mk( hoconsc:union([string(), map()])
```
Comment (JimMoen) [inline apps/emqx_authz/src/emqx_authz_api_schema.erl:78]: `position` should be unified with authn.
fix in other PR.

File: apps/emqx_authz/src/emqx_authz_api_schema.erl [hunk 4]
```diff
@@ -16,490 +16,185 @@
...
+
+to_list(A) when is_atom(A) ->
+    atom_to_list(A);
+to_list(B) when is_binary(B) ->
+    binary_to_list(B).
+
+to_bin(List) when is_list(List) ->
+    case io_lib:printable_list(List) of
+        true -> unicode:characters_to_binary(List);
+        false -> List
```
Comment (zhongwencool) [inline apps/emqx_authz/src/emqx_authz_api_schema.erl:197]: Let this crash? 
maybe we never enter this case?
we should always ensure the result is binary.

File: apps/emqx_authz/src/emqx_authz_api_schema.erl [hunk 5]
```diff
@@ -16,490 +16,185 @@
...
+    end;
+to_bin(Atom) when is_atom(Atom) -> atom_to_binary(Atom, utf8);
+to_bin(X) -> X.
```
Comment (zhongwencool) [inline apps/emqx_authz/src/emqx_authz_api_schema.erl:200]: same

File: apps/emqx_authz/src/emqx_authz_api_sources.erl [hunk 6]
```diff
@@ -489,8 +338,20 @@ do_write_file(Filename, Bytes) ->
            error(Reason)
     end.
 
-bin(Term) ->
-   erlang:iolist_to_binary(io_lib:format("~p", [Term])).
+bin(List) when is_list(List) ->
+    case io_lib:printable_list(List) of
+        true -> unicode:characters_to_binary(List);
+        false -> List
```
Comment (zhongwencool) [inline apps/emqx_authz/src/emqx_authz_api_sources.erl:344]: same

File: apps/emqx_authz/src/emqx_authz_api_schema.erl [hunk 7]
```diff
@@ -16,490 +16,185 @@
...
+
+transform_header_name(Headers) ->
+    maps:fold(fun(K0, V, Acc) ->
+                  K = list_to_binary(string:to_lower(to_list(K0))),
+                  maps:put(K, V, Acc)
+              end, #{}, Headers).
+
+%% body
+
+check_body(Body) ->
```
[the remark this answers is not in the document]
Comment (JimMoen) [reply apps/emqx_authz/src/emqx_authz_api_schema.erl:126]: Just a check. Same as `emqx_authn_http`
all body's value is bin with json decoded.

File: apps/emqx_authz/src/emqx_authz_api_schema.erl (hunk 1 above)
Comment (JimMoen) [reply apps/emqx_authz/src/emqx_authz_api_schema.erl:39]: Maybe change with `password-based` together in another PR.

**Review by zhongwencool — APPROVED** [review]

**Pull request merged**